src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -103,11 +107,17 @@ protected internal override void ClosePin(int pinNumber)
         {
             ValidatePinNumber(pinNumber);
 
-            if (_sysFSModes.ContainsKey(pinNumber) && _sysFSModes[pinNumber] == PinMode.Output)
+            lock (s_sysFsInitializationLock)
             {
-                Write(pinNumber, PinValue.Low);
-                SetPinMode(pinNumber, PinMode.Input);
+                if (_sysFSDriver != null)
+                {
+                    _sysFSDriver.ClosePin(pinNumber);
```

Comment:
I believe this is wrong as it could be the case that a pin is being used by the controller but it isn't open in the internal sysfs driver, so this would probably throw saying that we can't close a pin that isn't open.

Reply:
We could check our dictionary, but closing a pin that is not open is explicitly not an error.